tests/wallet/test_wallet_node.py

```diff
@@ -384,3 +388,65 @@ async def restart_with_fingerprint(fingerprint: Optional[int]) -> None:
     # Restart one more time and make sure the balance is still correct after start
     await restart_with_fingerprint(initial_fingerprint)
     assert await wallet_node.get_balance(wallet_id) == expected_more_balance
+
+
+@pytest.mark.asyncio
+async def test_race_cache_helper(root_path_populated_with_config: Path, get_temp_keyring: Keychain) -> None:
+    config = load_config(root_path_populated_with_config, "config.yaml", "wallet")
+    wallet_node = WalletNode(config, root_path_populated_with_config, test_constants, get_temp_keyring)
```

Comment:
in an ideal world the race cache would be a separate class that could be tested without constructing the full `WalletNode`. But this is fine for now I think

Reply:
I think we should get rid of this race cache anyway with the new protocol. Its just here because the messages come in the wrong order at the moment but we can't just change it.